source/tests/unit/ni_fake_service_tests.cpp

```diff
@@ -148,6 +175,39 @@ TEST(NiFakeServiceTests, NiFakeService_InitWithOptionsThenClose_SessionIsClosed)
   EXPECT_EQ(0, session_repository.access_session(session.id(), ""));
 }
 
+TEST(NiFakeServiceTests, NiFakeService_InitExtCalThenCloseExtCal_SessionIsClosed)
+{
+    nidevice_grpc::SessionRepository session_repository;
+    NiFakeMockLibrary library;
+    nifake_grpc::NiFakeService service(&library, &session_repository);
+    std::string session_name = "sessionName";
+    ViInt32 action = 1;
+    EXPECT_CALL(library, InitExtCal)
+        .WillOnce(DoAll(SetArgPointee<2>(kTestViSession), Return(kDriverSuccess)));
+    EXPECT_CALL(library, CloseExtCal(kTestViSession, action))
+        .WillOnce(Return(kDriverSuccess));
+
+    ::grpc::ServerContext context;
+    nifake_grpc::InitExtCalRequest init_request;
+    init_request.set_session_name(session_name);
+    nifake_grpc::InitExtCalResponse init_response;
+    ::grpc::Status init_status = service.InitExtCal(&context, &init_request, &init_response);
+    EXPECT_TRUE(init_status.ok());
+    nidevice_grpc::Session session = init_response.vi();
+    EXPECT_EQ(kTestViSession, session_repository.access_session(session.id(), ""));
+    EXPECT_EQ(kTestViSession, session_repository.access_session(0, session_name));
+    nifake_grpc::CloseExtCalRequest close_request;
+    close_request.mutable_vi()->set_id(session.id());
+    close_request.set_action(action);
+    nifake_grpc::CloseExtCalResponse close_response;
+    ::grpc::Status close_status = service.CloseExtCal(&context, &close_request, &close_response);
+
+    EXPECT_TRUE(close_status.ok());
+    EXPECT_EQ(kDriverSuccess, close_response.status());
+    EXPECT_NE(kTestViSession, session_repository.access_session(session.id(), ""));
+    EXPECT_EQ(0, session_repository.access_session(session.id(), ""));
```

Comment:
You meant to access session by name here, right?

Reply:
Done updating